week-1/Homework/extra/2-piping.js

```diff
@@ -1,58 +1,58 @@
-/*
-  PIPING FUNCTIONS
-  ================
-  1. Write 3 functions:
-  - one that adds 2 numbers together
-  - one that multiplies 2 numbers together
-  - one that formats a number so it's returned as a string with a £ sign before it (e.g. 20 -> £20)
-
-  2. Using the variable startingValue as input, perform the following operations using your functions all
-  on one line (assign the result to the variable badCode):
-  - add 10 to startingValue
-  - multiply the result by 2
-  - format it
-  
-  3. Write a more readable version of what you wrote in step 2 under the BETTER PRACTICE comment. Assign
-  the final result to the variable goodCode
-*/
-
-function add() {
-
+// PIPING FUNCTIONS
+//   ===
+//   === === === === =
+//   1. Write 3 functions:
+//   -one that adds 2 numbers together -
+//   one that multiplies 2 numbers together -
+//   one that formats a number so it’ s returned as a string with a£ sign before it(e.g.20 - > £20)
+// 2. Using the variable startingValue as input, perform the following operations using your functions all
+// on one line(assign the result to the variable badCode):
+//   -add 10 to startingValue -
+//   multiply the result by 2 -
+//   format it
+// 3. Write a more readable version of what you wrote in step 2 under the BETTER PRACTICE comment.Assign
+// the final result to the variable goodCode
+
+
+function add(num1, num2) {
+  const total = num1 + num2
+  return Number(total.toFixed(2))
```

Comment:
This is right, and works well, but one thing to think about is _when_ to round numbers. It's often useful to keep numbers as precise as we can during things like adding and multiplying, and only rounding them (with `.toFixed`) when we're about to show them to people. Otherwise we'll keep introducing more small rounding errors each time we do anything with numbers. So I'd think about removing the `Number` and `toFixed` calls here, and instead having one in the `format` function.

Like I say, this is right, and does what the question asks, but it's something to think about :)